### ISSUE TITLE
File marked as corrupt incorrectly (?)

### ISSUE THREAD

**Issue by kwhitefoot** [mono/taglib-sharp#182]
I'm trying to modify the keywords in a file but can't because the file is marked as corrupt when loading:

reason = "Size of entries exceeds possible data size"

Other tools including Windows 10 File Explorer are able to modify the keywords and taglib-sharp can read them.  What's happening?  Is this a bug or expected behaviour, that is, is this file really corrupt but other tools don't care?

![IMG_20190916_092218973_HDR](https://user-images.githubusercontent.com/183671/66487456-a41ab900-eaac-11e9-941a-d5872234c5f2.jpg)

The file is attached.

**Comment by JohnnyBravo75** [comment]
I have the same problem, many (most) of my pictures.
I can read the properties but I get "Size of entries exceeds possible data size" and cannot  save them (but other programs can)

**Comment by decriptor** [comment]
Interesting, it is very possible/likely a bug in taglib-sharp. Unfortunately, I'm short on time lately, but I'll see if I can't figure this out. Of course, patches are welcome ;)

If any of you have a very small image (size) that has this problem and don't mind I could include it in the test suite.

**Comment by JohnnyBravo75** [comment]
It seems to be a bug, and a serious one, because it prevents editing the metadata of a huge amount of images.
I can only include big images from camera (3-4mb).

**Comment by decriptor** [comment]
@JohnnyBravo75 Agreed. I have very limited bandwidth. I'll do my best to look at this.

**Comment by JohnnyBravo75** [comment]
![IMG_20191016_103202_small](https://user-images.githubusercontent.com/7339098/95855619-fabe7100-0d58-11eb-9bc8-0f0d3c122963.jpg)


` using (var fileMetaData = TagLib.File.Create(filePath, full ? TagLib.ReadStyle.Average : TagLib.ReadStyle.PictureLazy))
 {
    var allTags = (fileMetaData.Tag as CombinedImageTag);
 }`

**Comment by dbotz** [comment]
I am experiencing the same problem.  I noticed that if I remove the "makernotes" using exiftool (exiftool -makernotes:all=), taglib sharp is happy and does not mark the file corrupt.

**Comment by SSiefert** [comment]
I'm encountering the same issue. The first time I can change the properties - if I run the code the second time an error occurs "File is not writeable" and the corruption error is "Size of entries exceeds possible data size".

```
var file = TagLib.File.Create(filePath);

var imageTags = file.Tag as ImageTag;
if (imageTags == null) { continue; }

imageTags.Creator = "creator";
imageTags.Copyright = "copyright info;

imageTags.Keywords = null;
imageTags.ISOSpeedRatings = null;
imageTags.FocalLength = null;
imageTags.FocalLengthIn35mmFilm = null;
imageTags.Make = null;
imageTags.Model = null;
imageTags.Software = null;
imageTags.ExposureTime = null;
imageTags.FNumber = null;
imageTags.DateTime = null;
imageTags.Rating = null;
file.Save();
```

**Comment by Pascal-Krenckel** [comment]
I debugged it and indeed it is as @dbotz said, the problem is the makernote.
Every makernotes that is not recognized like the one from motorola in the first image are parsed as canon's makernotes. Obviously it fails and marks the file as corrupt.

**Comment by Shahid-khan5** [comment]
I am encountering the same error. Has anyone discovered a solution to this problem?

**Comment by Pascal-Krenckel** [comment]
If it is the same problem, either use another program to remove the makernote or add "return null;" at IFD/IFDReader.cs:893 and compile it yourself.

**Comment by Shahid-khan5** [comment]
ok, thanks. I will try

**Comment by Shahid-khan5** [comment]
Thanks @Pascal-Krenckel. It worked :)
